jupyter_server_proxy/__init__.py

```diff
@@ -1,5 +1,6 @@
 from jupyter_server.utils import url_path_join as ujoin
 
+from ._version import __version__  # noqa
```

Comment:
I guess once something _has_ been hoisted to `__init__.py`, for whatever reason, it is part of the public API. Doing in-the-wild searching sounds more tiresome.

This might be an appropriate time to consider getting on the recent push to get more of the stack working with `mypy`, with a backstop of the `unused-imports` rules from `ruff`, which can more generally be informed by [sp-repo-review](https://pypi.org/project/sp-repo-review/). This is likely better than making on-the-fly, case-by-case calls on specific repos.

Reply:
That sounds interesting @bollwyvl! I'm not able to drive that effort as part of this PR though, feeling somewhat overloaded atm :/